src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.LocalFunctions.cs

```diff
@@ -0,0 +1,252 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal partial class DataFlowPass
+    {
+        private readonly SmallDictionary<LocalFunctionSymbol, LocalFuncUsages> _localFuncVarUsages =
+            new SmallDictionary<LocalFunctionSymbol, LocalFuncUsages>();
+
+        private class LocalFuncUsages
+        {
+            public BitVector ReadVars = BitVector.Empty;
+            public BitVector WrittenVars = BitVector.Empty;
+
+            public bool LocalFuncVisited { get; set; } = false;
+        }
+
+        /// <summary>
+        /// At the local function's use site, checks that all variables read
+        /// are assigned and assigns all variables that are definitely assigned
+        /// to be definitely assigned.
+        /// </summary>
+        private void ReplayReadsAndWrites(LocalFunctionSymbol localFunc,
+                                          CSharpSyntaxNode syntax,
+                                          bool writes)
+        {
+            _usedLocalFunctions.Add(localFunc);
+
+            // If we're not inside a local function and the stateChangedAfterUse
+            // variable has been set then we don't want to report anything yet.
```

Comment:
Hang on. If you're not going to be replaying the data from the local functions, then you might as well not even scan the non-local-function statements in a block, because you're not helping the flow analysis computation converge faster (specifically, the definitely-assigned data at each label). I'll complete this thought at another place in the code.

---
In reply to: [75756034](https://github.com/dotnet/roslyn/pull/13234#discussion_r75756034) [](ancestors = 75756034,75743453)